dashboard/app/models/user.rb

```diff
@@ -1365,7 +1365,19 @@ def completed_progression_levels?(script)
     user_levels_by_level = user_levels_by_level(script)
 
     script.script_levels.none? do |script_level|
-      user_levels = script_level.level_ids.map {|id| user_levels_by_level[id]}
+      user_levels = []
+      script_level.levels.each do |level|
+        curr_user_level = user_levels_by_level[level.id]
+
+        # If level.id is not present in user_levels_by_level, check if level has contained_levels with present ids
+        if !curr_user_level && !level.contained_levels.empty?
```

Comment:
As I thought about this, I'm also wondering how this works with choice levels, such as https://studio.code.org/s/csc-timecapsule-2023/lessons/3/levels/2. In this case, I think we'd only want one level to be passing 🤔

Reply:
I tested locally and it looks like we create a user level for the BubbleChoice level itself, so I think we're good there! 🤞

Reply:
Oh great! Thank you for checking!